test/e2e/routes.test.ts

```diff
@@ -0,0 +1,128 @@
+import { describe, test, expect } from "./baseFixture"
+import { clean, tmpdir } from "../utils/helpers"
+import * as path from "path"
+import { promises as fs } from "fs"
+
+const routes = ["/", "/vscode", "/vscode/"]
+
+describe("VS Code Routes", ["--disable-workspace-trust"], {}, async () => {
+  const testName = "integrated-terminal"
+  test.beforeAll(async () => {
+    await clean(testName)
+  })
+
+  test("should load all route variations", async ({ codeServerPage }) => {
+    for (const route of routes) {
+      await codeServerPage.navigate(route)
+
+      // Check there were no redirections
+      const url = new URL(codeServerPage.page.url())
+      expect(url.pathname).toBe(route)
+
+      // Check that page loaded from correct route
+      const html = await codeServerPage.page.innerHTML("html")
+      switch (route) {
+        case "/":
+        case "/vscode/":
+          expect(html).toMatch(/src="\.\/[a-z]+-[0-9a-z]+\/static\//)
+          break
+        case "/vscode":
+          expect(html).toMatch(/src="\.\/vscode\/[a-z]+-[0-9a-z]+\/static\//)
+          break
+      }
+    }
+  })
+
+  test("should redirect to the passed in workspace using human-readable query", async ({ codeServerPage }) => {
+    const workspace = path.join(await tmpdir(testName), "test.code-workspace")
+    await fs.writeFile(workspace, "")
+
+    const url = new URL(codeServerPage.page.url())
+    expect(url.pathname).toBe("/")
+    expect(url.search).toBe(`?workspace=${workspace}`)
+  })
```

Comment:
This says it tests a passed-in workspace but it does not seem to pass it into the code-server invocation?  I have no idea how it passes. :stuck_out_tongue:

Reply:
Lol I think I accidentally wrote that test twice. Deleting one of them now.